src/NuGet.Clients/NuGet.CommandLine/MsBuildToolset.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            Path = GetMsBuildDirFromVsDir(sxsToolset.GetInstallationPath());
+            Version = GetMsBuildVersionFromMsBuildDir(Path);
+            _installDate = ConvertFILETIMEToDateTime(sxsToolset.GetInstallDate());
+        }
+
+        /// <summary>
+        /// This constructor is for testing purposes only
+        /// </summary>
+        public MsBuildToolset(string version, string path, DateTime installDate)
+        {
+            Version = version;
+            Path = path;
+            _installDate = installDate;
+        }
+
+        public Version ParsedVersion
+        {
+            get
+            {
+                if (_parsedToolsVersion == null)
+                {
+                    if (!System.Version.TryParse(Version, out _parsedToolsVersion))
+                    {
+                        _parsedToolsVersion = new Version(0, 0);
+                    }
+                }
+
+                return _parsedToolsVersion;
+            }
+        }
+
+        public string Version { get; private set; }
+
+        public string Path { get; private set; }
+
+        public int CompareTo(object obj)
+        {
+            if (Object.ReferenceEquals(obj, null))
+            {
+                return 1;
+            }
+
+            var rhs = obj as MsBuildToolset;
+            if (obj == null)
+            {
+                throw new ArgumentException("Comparison object not of correct type");
+            }
+
+            // Compare versions
+            var comparison = this.ParsedVersion.Major.CompareTo(rhs.ParsedVersion.Major);
+            if (comparison != 0)
+            {
+                return comparison;
+            }
+
+            comparison = this.ParsedVersion.Minor.CompareTo(rhs.ParsedVersion.Minor);
+            if (comparison != 0)
+            {
+                return comparison;
+            }
+
+            comparison = this.ParsedVersion.Build.CompareTo(rhs.ParsedVersion.Build);
+            if (comparison != 0)
+            {
+                return comparison;
+            }
+
+            comparison = this.ParsedVersion.Revision.CompareTo(rhs.ParsedVersion.Revision);
+            if (comparison != 0)
+            {
+                return comparison;
+            }
+
+            // Versions equal; compare by install date/time
+            return this._installDate > rhs._installDate ? 1 : -1;
+        }
+
+        private static DateTime ConvertFILETIMEToDateTime(FILETIME time)
+        {
+            long highBits = time.dwHighDateTime;
+            highBits = highBits << 32;
+            return DateTime.FromFileTimeUtc(highBits | (long)(uint)time.dwLowDateTime);
+        }
+
+        public static string GetMsBuildDirFromVsDir(string vsDir)
+        {
+            if (string.IsNullOrEmpty(vsDir))
+            {
+                return null;
+            }
+
+            string msBuildRoot = System.IO.Path.Combine(vsDir, "MSBuild");
```

Comment:
I'll need to use an alias as I have a property Path that gives me a conflict. But I don't want to change the property name as it's tidy. But yes, will do this--good call.